### ISSUE TITLE
Gazebo execution error

### ISSUE THREAD

**Issue by gmchoi96** [MOV-AI/movai-flow#237]
Have you checked our docs before proceeding?
- [ o ] [Documentation](https://flow.mov.ai/docs)
- [ o ] [Discourse](https://forum.flow.mov.ai/)

### Issue Description

When running Gazebo, the following error occurs:

[GUI] [Err] [Application.cc:757] [QT] Failed to create OpenGL context for format QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize 24, redBufferSize -1, greenBufferSize -1, blueBufferSize -1, alphaBufferSize -1, stencilBufferSize 8, samples -1, swapBehavior QSurfaceFormat::SwapBehavior(DoubleBuffer), swapInterval 1, profile  QSurfaceFormat::OpenGLContextProfile(NoProfile)) 

The nvidia driver was also installed, and rviz and mov.ai are running.

### Steps to Reproduce

  1. Thinking it was a temporary error, I tried both installation methods: optionA (source) & optionB (debian).
  2. I tried both host network mode and isolated network mode.
  3. However, gazebo does not run and an error appears.

#### Actual Behaviour

When entering the command movavi-flow-simulator, gazebo should run normally.
 I think it works.
Generally gazebo works fine, not mov.ai

#### Expected Behaviour

It seems that driver-related problems occur when running gazebo through docker.
This is because there is no driver problem when running Gazebo normally, and I think the problem is the same as the Gazebo problem that occurred when I installed the nvidia driver incorrectly before.

### Screenshots/GIFs

![image](https://github.com/MOV-AI/movai-flow/assets/149559257/843074ef-386a-4595-a094-8e3b7bc92e45)
![image](https://github.com/MOV-AI/movai-flow/assets/149559257/3c376c51-1253-4a2e-9934-39e277127578)
![image](https://github.com/MOV-AI/movai-flow/assets/149559257/e8980565-6e46-4a53-b463-2072e75d1c38)

### Versions

- movai-flow:
gmchoi@gmchoi:~$ apt list movai-flow 
Listing... Done
movai-flow/main,now 3.0.1-5 all [installed]
N: There is 1 additional version. Please use the '-a' switch to see it

![image](https://github.com/MOV-AI/movai-flow/assets/149559257/d154b513-95a8-40e3-9fce-eb40379a2bf4)

### Possible workarounds

- I think the driver version is messed up when running docker. If you have a similar solution, please explain.

### Additional Details

- ubuntu 22.04
- ROS2 humble

**Comment by MithunKinarullathil** [comment]
@gmchoi96 Thank you for the report. We are looking into it.

**Comment by MithunKinarullathil** [comment]
@gmchoi96 Can you try to do this command (`xhost +local:docker >/dev/null 2>&1
`) on the terminal and launch the simulator using `movai-flow-simulator` command please?

**Comment by AlexFernandes-MOVAI** [comment]
@gmchoi96 I think your MOVAI Flow installation is all good, we have positive traces showing `NVIDIA GPU card found`, `NVIDIA Container Toolkit found` and the container itself is starting so, everything seems to be well detected and working at the driver level at this step.

However, if you have done and redone the installation of the drivers after starting MOVAI Flow, I would recommend you to reboot your host (yes only way to fix some issues about lib and drivers mismatch), recreate the containers with `movai-flow-reset` and then follow the steps of @MithunKinarullathil:

```
xhost +local:docker
movai-flow-launch
movai-flow-simulator
```

Finally, note that you can also have hit this [issue](https://askubuntu.com/questions/1412683/ubuntu-20-04-libgl-error-failed-to-load-driver-swrast) or similar where CUDA, libmesa and nvidia-drivers are conflicting and prevent QT and OpenGL to work correctly...

**Comment by gmchoi96** [comment]
@MithunKinarullathil @AlexFernandes-MOVAI 
Thanks for replying

Unfortunately, it didn't work out the way you suggested.
1. `movai-flow-reset`
2. `xhost +local:docker >/dev/null 2>&1`
3. `movai-flow-launch`
4. `movai-flow-simulator`

the error was the same


[issue](https://askubuntu.com/questions/1412683/ubuntu-20-04-libgl-error-failed-to-load-driver-swrast)
About the given link
This is a newly installed driver that also removed the nvidia driver provided by default in Ubuntu.

additionally
Is there any need to modify the docker_compose.yml file?

I saw a post saying it was solved by editing the file... but I couldn't try it because the process wasn't detailed.

**Comment by MithunKinarullathil** [comment]
> @MithunKinarullathil @AlexFernandes-MOVAI Thanks for replying
> 
> Unfortunately, it didn't work out the way you suggested.
> 
> 1. `movai-flow-reset`
> 2. `xhost +local:docker >/dev/null 2>&1`
> 3. `movai-flow-launch`
> 4. `movai-flow-simulator`
> 
> the error was the same
> 
> [issue](https://askubuntu.com/questions/1412683/ubuntu-20-04-libgl-error-failed-to-load-driver-swrast) About the given link This is a newly installed driver that also removed the nvidia driver provided by default in Ubuntu.
> 
> additionally Is there any need to modify the docker_compose.yml file?

No, there isn't. We tested in several PCs and it worked without any modifications.
> 
> I saw a post saying it was solved by editing the file... but I couldn't try it because the process wasn't detailed.

If you could provide the link that would be helpful. We currently are not able to figure out the issue. If you have solved the issue, please feel free to open a PR.
